rdr_service/dao/biobank_specimen_dao.py

```diff
@@ -0,0 +1,98 @@
+                                                    ('confirmationDate', 'confirmedDate'),
+                                                    ('studyID', 'studyId'),
+                                                    ('repositoryID', 'repositoryId'),
+                                                    ('rlimsID', 'rlimsId'),
+                                                    ('testcode', 'testCode')]:
+            result[client_field_name] = result.pop(model_field_name)
+
+        # Translate biobankId
+        result['participantID'] = to_client_biobank_id(result.pop('biobankId'))
+
+        # Format dates
+        for date_field in ['collectionDate', 'confirmationDate', 'processingCompleteDate', 'created']:
+            format_json_date(result, date_field)
+
+        result_status = result['status']
+        result['status'] = {}
+        for status_field in ['deviations', 'freezeThawCount', 'location', 'processingCompleteDate', 'quantity',
+                             'quantityUnits']:
+            if status_field in result:
+                result['status'][status_field] = result.pop(status_field)
+        result['status']['status'] = result_status
+
+        return result
+
+    #pylint: disable=unused-argument
+    def from_client_json(self, resource, id_=None, expected_version=None, participant_id=None, client_id=None):
+        order = BiobankSpecimen(rlimsId=resource['rlimsID'], orderId=resource['orderID'], testCode=resource['testcode'],
+                                biobankId=from_client_biobank_id(resource['participantID']))
+
+        if not self.exists(resource):
+            order.created = clock.CLOCK.now()
+
+        for client_field, model_field, parser in [('repositoryID', 'repositoryId', None),
+                                                  ('studyID', 'studyId', None),
+                                                  ('cohortID', 'cohortId', None),
+                                                  ('sampleType', 'sampleType', None),
+                                                  ('collectionDate', 'collectionDate', parse_date),
+                                                  ('confirmationDate', 'confirmedDate', parse_date)]:
+            self.map_optional_json_field_to_object(resource, order, client_field, model_field, parser)
+
+        if 'status' in resource:
+            for status_field_name, parser in [('status', None),
+                                              ('freezeThawCount', None),
+                                              ('location', None),
+                                              ('quantity', None),
+                                              ('quantityUnits', None),
+                                              ('deviations', None),
+                                              ('processingCompleteDate', parse_date)]:
+                self.map_optional_json_field_to_object(resource['status'], order, status_field_name, parser=parser)
```

Comment:
It seems you only handled part of the payload. In the swagger API description, the full payload is something like below:
```
{
  "rlimsID": "string",
  "participantID": "string",
  "orderID": "string",
  "testcode": "string",
  "repositoryID": "string",
  "studyID": "string",
  "cohortID": "string",
  "sampleType": "string",
  "status": {
    "status": "string",
    "freezeThawCount": 0,
    "location": "string",
    "quantity": "string",
    "quantityUnits": "string",
    "processingCompleteDate": "2020-05-05T19:40:28.534Z",
    "deviations": "string"
  },
  "disposalStatus": {
    "reason": "string",
    "disposalDate": "2020-05-05T19:40:28.534Z"
  },
  "collectionDate": "2020-05-05T19:40:28.534Z",
  "confirmationDate": "2020-05-05T19:40:28.534Z",
  "attributes": [
    {
      "name": "string",
      "value": "string"
    }
  ],
  "aliquots": [
    {
      "rlimsID": "string",
      "sampleType": "string",
      "status": {
        "status": "string",
        "freezeThawCount": 0,
        "location": "string",
        "quantity": "string",
        "quantityUnits": "string",
        "processingCompleteDate": "2020-05-05T19:40:28.534Z",
        "deviations": "string"
      },
      "disposalStatus": {
        "reason": "string",
        "disposalDate": "2020-05-05T19:40:28.534Z"
      },
      "childPlanService": "string",
      "initialTreatment": "string",
      "containerTypeID": "string",
      "datasets": [
        {
          "rlimsID": "string",
          "name": "string",
          "status": "string",
          "datasetItems": [
            {
              "paramID": "string",
              "displayValue": "string",
              "displayUnits": "string"
            }
          ]
        }
      ],
      "aliquots": [
        null
      ]
    }
  ]
}
```

Reply:
Good point. I understood them to be part of the implementation for the other endpoints. For example, I thought implementation of handling disposalStatus would be part of the 4th endpoint listed in the tech design document.